src/client.rs

```diff
@@ -91,9 +119,12 @@ impl<Req, Resp, E> fmt::Debug for Client<Req, Resp, E>
 }
 
 /// Additional options to configure how the client connects and operates.
-#[derive(Clone, Default)]
+#[cfg_attr(feature = "tls", derive(Default))]
+#[cfg_attr(not(feature = "tls"), derive(Clone, Default))]
```

Comment:
Let's just have it not be `Clone`. I don't feel super comfortable with the traits impl'd changing based on which set of features are enabled.

cc @shaladdle